continuous-integration/check.sh

```diff
@@ -0,0 +1,248 @@
+#!/bin/bash -e
```

Comment:
Small nitpick, but could we name this file something like `test/travis-ci.sh` so that it's clear that it isn't intended to do anything on your local machine?